File: src/main/java/virtualMachine/stack/vm_instruction_parsing/vm_instruction_processing/impl/PushInstructionProcessor.java
package virtualMachine.stack.vm_instruction_parsing.vm_instruction_processing.impl;

import virtualMachine.stack.memory.VirtualMemory;
import virtualMachine.stack.types.instruction.Instruction;
import virtualMachine.stack.vm_instruction_parsing.vm_instruction_processing.InstructionProcessor;

import static virtualMachine.stack.memory.MemorySegments.getFromName;

public class PushInstructionProcessor implements InstructionProcessor {
    @Override
    public void processInstruction(Instruction instruction, VirtualMemory virtualMemory) {
        var memoryAddress = instruction.getNumericValue();
        var segment = getFromName(instruction.getOperand());

        var loadedVariable = virtualMemory.getFromMemory(memoryAddress, segment);
        virtualMemory.push(loadedVariable);
    }
}

File: src/main/kotlin/virtualMachine/stack/vm_instruction_parsing/vm_instruction_processing/impl/PushInstructionProcessor.kt
package virtualMachine.stack.vm_instruction_parsing.vm_instruction_processing.impl

import virtualMachine.stack.memory.GlobalVirtualMemory
import virtualMachine.stack.datawrappers.SixteenBit
import virtualMachine.stack.memory.MemorySegments
import virtualMachine.stack.vm_instruction_parsing.vm_instruction_processing.InstructionProcessor

class PushInstructionProcessor : InstructionProcessor {

    override fun processInstruction(instruction: String, virtualMemory: GlobalVirtualMemory) {

        val splitCommand: List<String> = instruction.split(" ")
        val index = splitCommand[2].toInt()
        val segment = MemorySegments.getFromName(splitCommand[1])

        val storedVariable : SixteenBit = virtualMemory.getFromMemory(index, segment)
        virtualMemory.pushToLocalStack(storedVariable)
    }

}